src/marqo/tensor_search/tensor_search.py

```diff
@@ -2648,3 +2691,33 @@ def delete_documents(config: Config, index_name: str, doc_ids: List[str]):
             document_ids=doc_ids,
         )
     )
+
+def normalize_vector(embeddings: Union[List[List[float]], ndarray]) -> List[List[float]]:
+    """
+    Normalizes a list of vectors (embeddings) to have unit length.
+
+    Args:
+        embeddings (Union[List[List[float]], ndarray]): A list of vectors or a numpy ndarray of vectors to be normalized.
+
+    Returns:
+        List[List[float]]: A list of normalized vectors.
+    """
+
+    # Convert the input embeddings to a numpy array
+    if embeddings.__class__ == ndarray:
+        embeddings_array = embeddings
+    else:
+        embeddings_array = np.array(embeddings)
+
+    # Calculate the magnitude (Euclidean norm) of each vector along the last axis
+    magnitude = np.linalg.norm(embeddings_array, axis = -1, keepdims=True)
```

Comment:
Why not reuse the normalization code?

Reply:
Not sure what you mean here

Reply:
Checked offline with @farshidz and confirmed that this is not applicable